core/docker/importer.sh

```diff
@@ -1,2 +1,23 @@
+#!/bin/sh
 
-curl -X POST --insecure -u "$OPENHIM_API_USERNAME:$OPENHIM_API_PASSWORD" -H "Content-Type: application/json" -d @openhim-import.json "$OPENHIM_API_SERVER/metadata" -v
\ No newline at end of file
+OPENHIM_RESPONSE="";
+TIMEOUT=0
+
+while [ "$OPENHIM_RESPONSE" != "200" ]
+do
+  TIMEOUT=$((TIMEOUT+1))
+  # Fail script if incomplete at 30 seconds
+  if [ "$TIMEOUT" == "15" ]
+  then
+    echo -e "Sleep timeout reached: $TIMEOUT.\nFailed to import config. Could not reach the OpenHIM."
+    exit 1
+  fi
+
+  echo "OpenHIM not ready ( $OPENHIM_RESPONSE ) - sleeping"
+  sleep 2
+  OPENHIM_RESPONSE=$(curl -X GET --insecure --write-out %{http_code} --silent --output /dev/null "$OPENHIM_API_SERVER/heartbeat");
+done
```

Comment:
This looks like it would work well, however, we might use this same script to configure a number of services. I wonder if we should include a utility like this - https://www.npmjs.com/package/wait-on - in our `Dockerfile.importer` and rather use that to wait-on all services to be available. Have a look at the tool about and let me know your thoughts.

Reply:
I'm sure similar tools exist if you want to try find a bash only one, but node doesn't add much weight to the image.

Reply:
Thanks @rcrichton Thats a good idea. We could make use of the `node-alpine` base docker image to keep things small. Maybe we should be changing this importer script to rather be a node application which will handle the importing logic? This could also easiliy be updated to cater for other import methods

What do you think? Reason for initially using bash is its quick and easy to get something basic working. More complex flows are def possible but can become quite confusing to follow and maintain.

Reply:
Yeah, transitioning this to a js script might be good. I don't have a problem between bash or js really. Either is a good option. We could just use the wait-on CLI in a bash script.

js might be easier for us to maintain so that is a plus.